java/pgjdbc_using_dsql_connector/README.md

```diff
@@ -0,0 +1,191 @@
+
+2. **CustomCredentialsProviderExample** - Shows custom AWS credentials:
+   - Using AWS profile credentials
+   - STS assume role credentials
+   - Custom credentials provider implementation
+
+## Supported URL Formats
+
+The Aurora DSQL JDBC Connector supports the following URL format:
+
+1. **jdbc:aws-dsql:postgresql**: `jdbc:aws-dsql:postgresql://your-cluster.dsql.us-east-1.on.aws/postgres?user=admin`
+
+## ⚠️ Important
+
+* Running this code might result in charges to your AWS account.
+* We recommend that you grant your code least privilege. At most, grant only the
+  minimum permissions required to perform the task. For more information, see
+  [Grant least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege).
+* This code is not tested in every AWS Region. For more information, see
+  [AWS Regional Services](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services).
+
+## Run the examples
+
+### Prerequisites
+
+* You must have an AWS account, and have your default credentials and AWS Region
+  configured as described in the
+  [Globally configuring AWS SDKs and tools](https://docs.aws.amazon.com/credref/latest/refdocs/creds-config-files.html)
+  guide.
+* Java Development Kit (JDK): Ensure you have JDK 17+ installed.
+
+   _To verify Java is installed, you can run:_
+   ```bash
+   java -version
+   ```
+
+* Build Tool: Gradle is used for this project.
+   - The project includes a Gradle connector, so no separate Gradle installation is required.
+
+* AWS SDK: The project includes the latest version of the AWS Java SDK dependencies.
+
+* Aurora DSQL JDBC Connector: The project includes the latest version of the AWS Aurora DSQL JDBC Connector dependencies
+
+* You must have an Aurora DSQL cluster. For information about creating an Aurora DSQL cluster, see the
+  [Getting started with Aurora DSQL](https://docs.aws.amazon.com/aurora-dsql/latest/userguide/getting-started.html)
+  guide.
+
+* If connecting as a non-admin user, ensure the user is linked to an IAM role and is granted access to the appropriate
+  schema. See the
+  [Using database roles with IAM roles](https://docs.aws.amazon.com/aurora-dsql/latest/userguide/using-database-and-iam-roles.html)
+  guide.
+
+### Environment Variables
+
+Set the following environment variables for your cluster details:
+
+```bash
+# Required: Aurora DSQL cluster endpoint
+export CLUSTER_ENDPOINT="<your-cluster-endpoint>"
+# e.g., "foo0bar1baz2quux3quuux4.dsql.us-east-1.on.aws"
+
+# Required: Aurora DSQL cluster user (e.g., "admin")
+export CLUSTER_USER="<your-user>"
+
+# Optional: AWS profile name for custom credentials
+export PROFILE="<your-aws-profile>"
+
+# Optional: IAM role ARN for assume role credentials
+export ROLE_ARN="<your-role-arn>"
+```
+
+### Run the code
+
+You can use the provided setup script to configure environment variables:
+
+```bash
+# Set up environment variables interactively
+./setup-env.sh
```

Comment:
Please add GitHub workflow steps for each of these samples to verify they work properly on the CI system.